### PR TITLE
Updated surface tracker documentation. 

### LINKED ISSUE #227: Add docs for surface tracker revamp
PR: https://github.com/pupil-labs/pupil/pull/1268
Assigned to @marc-tonsen

### PR DESCRIPTION
**Pull request by marc-tonsen**
Fixes #227. Changes are based on surfaces revamp PR #1268 in Pupil and should be merged together.

### CODE REVIEW

**Comment by willpatera** [comment]
@marc-tonsen I will keep this open until https://github.com/pupil-labs/pupil/pull/1268 is merged.

**Comment by willpatera** [comment]
@marc-tonsen @papr I would like to merge, but please resolve conflicts. There would be redundant information if we merged this PR.

**Comment by marc-tonsen** [comment]
Conflicts are resolved now @willpatera !

File: user-docs/pupil-capture.md [hunk 1]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+*   Click the `edit surface` button and move the corners of your surface into the desired position.
+    In the online case this is much easier if you freeze the video by clicking the `Freeze Scene` button.
+*   If markers have been erroneously added or left out, click the `add/remove markers` button and afterwards onto the according marker to add/remove them from your surface.
+
+
+#### Reusing Surface Definitions
+
+Your surfaces are automatically saved in a file called `surface_definitions` in the `pupil_capture_settings` directory. 
+If you restart Pupil Capture or the Surface Tracker plugin, your surface definitions from previous sessions will be loaded.
+Further the `surface_definitions` file is copied into each recording folder as well, so you have access to your surface definitions in Pupil Player.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
The `surface_definitions` file is copied into each recording folder as well, so you will have access to your surface definitions in Pupil Player. 
```

File: user-docs/pupil-capture.md [hunk 2]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+In both cases the necessary steps are as follows:
+
+*   Prepare your environment as described above.
+*   Turn on the `Surface Tracker` plugin .
+*   Make sure the camera is pointing at your AOI and the markers are well detected.
+    In the offline case seek to a frame that contains a good view of your desired AOI.
+*   Add a new surface by clicking the `Add surface` button.
+*   Give your surface a name.
+*   Click the `edit surface` button and move the corners of your surface into the desired position.
+    In the online case this is much easier if you freeze the video by clicking the `Freeze Scene` button.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
    In the real-time case (using Pupil Capture) this is much easier if you freeze the video by clicking the `Freeze Scene` button.
```

File: user-docs/pupil-capture.md [hunk 3]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+
+> {{< video-webm src="/videos/surface-tracking/srf-tracking.webm" >}}
+
+Surfaces can be defined with Pupil Capture in real-time, or offline with Pupil Player. 
+In both cases the necessary steps are as follows:
+
+*   Prepare your environment as described above.
+*   Turn on the `Surface Tracker` plugin .
+*   Make sure the camera is pointing at your AOI and the markers are well detected.
+    In the offline case seek to a frame that contains a good view of your desired AOI.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
    In the post-hoc case (using Pupil Player) seek to a frame that contains a good view of your desired AOI.
```

File: user-docs/pupil-capture.md [hunk 4]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
-*  Generate markers with [this script](https://github.com/pupil-labs/pupil-helpers/blob/master/markers_stickersheet/make_square_markers.py), or download the image.
 
-<aside class="notice">
-  <strong>Note</strong> - When printing markers, ensure that white space remains around the square marker. You can scale the markers to different sizes, but make sure to have a white border width of at least 1.2 x the marker grid size for marker, unless the marker is affixed onto a white (or light colored) background.
-</aside>
+#### Defining a Surface 
+
+> {{< video-webm src="/videos/surface-tracking/srf-tracking.webm" >}}
+
+Surfaces can be defined with Pupil Capture in real-time, or offline with Pupil Player.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
Surfaces can be defined with Pupil Capture in real-time, or post-hoc with Pupil Player. 
```

File: user-docs/pupil-capture.md [hunk 5]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
-*  We have created a window that shows registered surfaces within the world view and the gaze positions that occur within those surfaces in realtime.
-*  Streaming Surfaces with Pupil Capture - Detected surfaces as well as gaze positions relative to the surface are broadcast under the `surface` topic. Check out [this video](http://youtu.be/qHmfMxGST7A) for a demonstration.
-*  Surface Metrics with Pupil Player - if you have defined surfaces, you can generate surface visibility reports or gaze count per surface. See our [blog post](http://pupil-labs.com/blog/2014/07/0392-player-release.html) for more information.
+A surface can be based on one or more markers.
+The markers need to be placed in close proximity or within your desired AOI.
+If your AOI is for example a computer monitor, you could display your markers in the corners of the screen or place them somewhere on the bezel.
+If your AOI is a magazine page you could place the markers in the corners of the page or anywhere else on the page where they are not occluding.
+When placing your markers please follow the following rules:
 
+*   All markers of a surface need to lie within the same plane.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
*   All markers of a surface need to lie within the same two dimensional plane.
```

File: user-docs/pupil-capture.md [hunk 6]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
-*  We have created a window that shows registered surfaces within the world view and the gaze positions that occur within those surfaces in realtime.
-*  Streaming Surfaces with Pupil Capture - Detected surfaces as well as gaze positions relative to the surface are broadcast under the `surface` topic. Check out [this video](http://youtu.be/qHmfMxGST7A) for a demonstration.
-*  Surface Metrics with Pupil Player - if you have defined surfaces, you can generate surface visibility reports or gaze count per surface. See our [blog post](http://pupil-labs.com/blog/2014/07/0392-player-release.html) for more information.
+A surface can be based on one or more markers.
+The markers need to be placed in close proximity or within your desired AOI.
+If your AOI is for example a computer monitor, you could display your markers in the corners of the screen or place them somewhere on the bezel.
+If your AOI is a magazine page you could place the markers in the corners of the page or anywhere else on the page where they are not occluding.
+When placing your markers please follow the following rules:
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
When placing your markers please follow the guidelines:
```

File: user-docs/pupil-capture.md [hunk 7]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
-*  Surfaces defined with more than 2 markers are detected even if some markers go outside the field of vision or are obscured.
-*  We have created a window that shows registered surfaces within the world view and the gaze positions that occur within those surfaces in realtime.
-*  Streaming Surfaces with Pupil Capture - Detected surfaces as well as gaze positions relative to the surface are broadcast under the `surface` topic. Check out [this video](http://youtu.be/qHmfMxGST7A) for a demonstration.
-*  Surface Metrics with Pupil Player - if you have defined surfaces, you can generate surface visibility reports or gaze count per surface. See our [blog post](http://pupil-labs.com/blog/2014/07/0392-player-release.html) for more information.
+A surface can be based on one or more markers.
+The markers need to be placed in close proximity or within your desired AOI.
+If your AOI is for example a computer monitor, you could display your markers in the corners of the screen or place them somewhere on the bezel.
+If your AOI is a magazine page you could place the markers in the corners of the page or anywhere else on the page where they are not occluding.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
If your AOI is a magazine page, you could place the markers in the corners of the page, or anywhere else on the page where they are not occluding the content.
```

File: user-docs/pupil-capture.md [hunk 8]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+    In the online case this is much easier if you freeze the video by clicking the `Freeze Scene` button.
+*   If markers have been erroneously added or left out, click the `add/remove markers` button and afterwards onto the according marker to add/remove them from your surface.
+
+
+#### Reusing Surface Definitions
+
+Your surfaces are automatically saved in a file called `surface_definitions` in the `pupil_capture_settings` directory. 
+If you restart Pupil Capture or the Surface Tracker plugin, your surface definitions from previous sessions will be loaded.
+Further the `surface_definitions` file is copied into each recording folder as well, so you have access to your surface definitions in Pupil Player. 
+You can copy&paste this file to move definitions from one session or recording to another.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
You can copy & paste this file to move definitions from one session or recording to another.
```

File: user-docs/pupil-capture.md [hunk 9]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+
 
-#### Moving Surface Definitions
+#### Gaze Heatmaps for Surfaces
 
-The surfaces are saved in a file is called `surface_definitions`. This file is in every `pupil_capture_settings` directory and in every recording directory. You can copy&paste this file to move definitions from one session or recording to another.
+It is possible to display gaze heatmaps for each surface by enabling `Show Heatmap` in the `Surface Tracker` menu.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
You can display gaze heatmaps for each surface by enabling `Show Heatmap` in the `Surface Tracker` menu.
```

File: user-docs/pupil-capture.md [hunk 10]
```diff
@@ -346,48 +346,87 @@ to start and stop recordings that are stored in the phone.
...
+Two heatmap modes are supported:
+*   `Gaze within each surface`: Visualizes the distribution of gaze points that lie within each surface. 
+*   `Gaze across different surfaces`: Color codes the surfaces to visualize the amount of time spend gazing on each surface in relation to other surfaces.
+Red color stands for a lot of gaze points or time spend, while blue color stands for few gaze points or little time spend.
```
Comment (willpatera) [inline user-docs/pupil-capture.md]:
```suggestion
Red color represents a lot of gaze points or time spent. Blue color represents few gaze points or little time spent. 
```

**Comment by marc-tonsen** [comment]
@willpatera I accepted all your suggestions.

**Pull request merged**